relentless/optimize/objective.py

```diff
@@ -165,3 +181,150 @@ def design_variables(self):
+    and :math:`\rho=N/V` is the number density, all for the target ensemble.
+
+    Parameters
+    ----------
+    target : :class:`~relentless.ensemble.Ensemble`
+        The target ensemble (must have specified N and V).
+    simulation :class:`~relentless.simulate.Simulation`
+        The simulation engine to use, with specified simulation operations.
+    potentials :class:`~relentless.simulate.Potentials`
+        The pair potentials to use in the simulations.
+    thermo : :class:`~relentless.simulate.SimulationOperation`
+        The thermodynamic analyzer for the simulation ensemble and rdf (usually
+        :meth:`AddEnsembleAnalyzer()` for the specified simulation type.
+    dr : float
+        The radial step size.
+    communicator : :class:`~relentless.mpi.Communicator`
+        The communicator used for saving outputs to file ##?## (defaults to
+        ``None``).
+
+    """
+    def __init__(self, target, simulation, potentials, thermo, dr, communicator=None):
+        self.target = target
+        self.simulation = simulation
+        self.potentials = potentials
+        self.thermo = thermo
+        self.dr = dr
+        self.communicator = communicator
+
+    def compute(self, directory=None):
+        """Evaluate the value and gradient of the relative entropy function.
+
+
+        Parameters
+        ----------
+        directory : :class:`~relentless.data.Directory`
+            The directory to which the values of the pair potential design
+            variables at the time of computation are saved (defaults to ``None``).
+
+        Returns
+        -------
+        :class:`ObjectiveFunctionResult`
+            The result of :meth:`ObjectiveFunction.make_result()`
+
+        """
+        sim = self.simulation.run(self.target, self.potentials, directory, self.communicator)
+        sim_ens = self.thermo.extract_ensemble(sim)
+
+        # compute the relative entropy gradient by integration
+        g_tgt = self.target.rdf
+        g_sim = sim_ens.rdf
+        dvars = self.design_variables()
+        gradient = _collections.KeyedArray(keys=dvars)
+
+        for var in dvars:
+            update = 0
+            for i,j in self.target.pairs:
+                rs = self.potentials.pair.r
+                us = self.potentials.pair.energy((i,j))
+                dus = self.potentials.pair.derivative((i,j),var)
+
+                #only count (continuous range of) finite values
+                flags = np.isinf(us)
+                first_finite = 0
+                while not flags[first_finite] and first_finite < len(rs):
+                    first_finite += 1
+                rs = rs[first_finite+1:]
+                dus = dus[first_finite+1:]
+
+                #interpolate derivative wrt design variable with r
+                dudvar = _math.Interpolator(rs,dus)
+
+                # find common domain to compare rdfs
+                r0 = max(g_sim[i,j].table[0][0],g_tgt[i,j].table[0][0])
```

Comment:
Also need `rs[0]` here, since it could be bigger than either of these.
```suggestion
                r0 = max(g_sim[i,j].table[0,0],g_tgt[i,j].table[0,0],dudvar.table[0,0])
```